From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Marc Ferland <marc.ferland@gmail.com>
Date: Fri, 5 Sep 2026 12:00:00 -0400
Subject: [PATCH] Avoid absolute paths in generated parser sources

Disable Flex and Bison line directives so their absolute input and output
paths are not embedded in generated sources.

Upstream-Status: Inappropriate [oe-specific]

SPDX-FileCopyrightText: Marc Ferland <marc.ferland@sonatest.com>
SPDX-License-Identifier: MIT

Signed-off-by: Marc Ferland <marc.ferland@gmail.com>
---
 src/common/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index c58caf2..963f7e5 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -10,11 +10,11 @@ set(common_src
 if(HAVE_SURVEY_TARGET_EXPRESSIONS)

 if (CMAKE_VERSION VERSION_LESS 3.5.0)
-    set(_flex_opts COMPILE_FLAGS --nounistd --header-file=${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionscanner.h)
-    set(_bison_opts COMPILE_FLAGS --defines=${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionparser_p.h)
+    set(_flex_opts COMPILE_FLAGS "--nounistd --noline" --header-file=${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionscanner.h)
+    set(_bison_opts COMPILE_FLAGS "--no-lines" --defines=${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionparser_p.h)
 else()
-    set(_flex_opts COMPILE_FLAGS --nounistd DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionscanner.h)
-    set(_bison_opts DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionparser_p.h)
+    set(_flex_opts COMPILE_FLAGS "--nounistd --noline" DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionscanner.h)
+    set(_bison_opts COMPILE_FLAGS "--no-lines" DEFINES_FILE ${CMAKE_CURRENT_BINARY_DIR}/surveytargetexpressionparser_p.h)
 endif()

 flex_target(surveytargetexpressionscanner
--
2.53.0
